### ISSUE TITLE
cmd/cue: allow codec to be specified when importing with `cue get go`

### ISSUE THREAD

**Issue by rogpeppe** [cue-lang/cue#2264]
In Go code, the encoding of a struct depends on what code it's encoded with.

```
type T struct {
	Foo int
	Bar string `json:"b"`
	Cod int    `yaml:"c"`
}
```

Here's a demonstration in the Go playground: https://go.dev/play/p/jC_BbaZpshw

In CUE terms, the above can legitimately encode either to this (when using the json or k8s yaml encoder):

```
Foo: int64
b: string
Cod: int64
```

or this (when using the gopkg.in yaml encoder):

```
foo: int64
bar: string
c: int64
```

The `cue` package chooses neither of these, but instead encodes the type as:

```
Foo: int64
b:   string
c:   int64
```

That is, it picks and chooses which kind of struct tag to use based only on that
struct tag in isolation.

I would suggest that it should be possible to select an encoder type when importing
Go code to cater for situations where `gopkg.in/yaml` is used to marshal Go structs.
For example:
```
cue get go --codec yaml $pkg
```
By default, the importer should be consistent in choosing names from `json` struct tags only.

See also #2261.

**Comment by mvdan** [comment]
Per chat with @mpvl and @myitcv, we think it would be a nice transition to use `cue fix` to rewrite any fields using byte literals to using string literals instead. This should work fine for any such fields which use valid UTF-8 values. Any which aren't valid UTF-8 can be an error, which would surface any potentially valid use case we would be breaking. We could fully remove support for bytes as fields in a later release.

**Comment by myitcv** [comment]
This might be a duplicate of https://github.com/cue-lang/cue/issues/1026?

**Comment by mvdan** [comment]
I don't think this is a dupe of #1026; that relates to `inline` and embed semantics, and this is about which encoding tags to use or prioritize. They are related, but not duplicates.

I agree with @rogpeppe's assessment above; when encoding or decoding a Go type, it doesn't make sense to mix JSON and YAML tags. However, I disagree with switching the default to choosing `json` struct field tags alone.

Today, if I `cue get go` Go types which only have `yaml` struct field tags, those are picked up. We should continue to do that, at least for one full CUE release, for the sake of smoothing out the transition for end users. So, for example, the default for v0.16.0 could be `--codec json,yaml` and for v0.17.0 we could switch the default to `--codec json`. The comma-separated `json,yaml` form would be equivalent to what we do today - use the JSON tag if present, and if not the YAML tag.

I also wonder if perhaps a global CLI flag might not be enough, e.g. if different packages use different encodings. But perhaps we look at that if a user raises such a concern in the future.
